compiler/src/codegen.rs

```diff
@@ -40,19 +43,50 @@ impl Default for TagEnvironment {
 
 pub trait Backend: Default {
     fn tag_environment(&mut self, environment: TagEnvironment);
-    fn generate_type(&mut self, ty: &Type) -> Result<String>;
+    fn generate_type(
+        &mut self,
+        ty: &Type,
+        parent_prefix: Option<&Prefix>,
+    ) -> Result<(String, Option<Prefix>)>;
```

Comment:
I HATE this too, but for prefixed of fields, like `someFieldWithContextNumber [2] INTEGER`, the `[2] INTEGER` is parsed here but the caller has already written the field name and we can't add a macro call here, so we need to return extra data to caller, so it can write the macro call in the right place.